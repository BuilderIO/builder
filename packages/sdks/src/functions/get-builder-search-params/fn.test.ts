import { describe, test, expect } from '@jest/globals';
import { getBuilderSearchParams, convertSearchParamsToQueryObject } from '.';

const querystring =
  'someotherValue=jklsjfdal&abc=klfdjklgfds&builder.cachebust=true&builder.preview=page&builder.noCache=true&__builder_editing__=true&builder.overrides.page=037948e52eaf4743afed464f02c70da4&builder.overrides.037948e52eaf4743afed464f02c70da4=037948e52eaf4743afed464f02c70da4&builder.overrides.page%3A%2F=037948e52eaf4743afed464f02c70da4&preview_theme_id=128854393017';

const url = new URL(`localhost:3000/about-us?${querystring}`);

describe('Get Builder SearchParams', () => {
  test('correctly converts URLSearchParams to object', () => {
    const output = convertSearchParamsToQueryObject(url.searchParams);
    expect(output).toMatchSnapshot();
  });
  test('correctly extracts all builder params from a query object', () => {
    const output = getBuilderSearchParams(
      convertSearchParamsToQueryObject(url.searchParams)
    );
    expect(output).toMatchSnapshot();
  });
});
