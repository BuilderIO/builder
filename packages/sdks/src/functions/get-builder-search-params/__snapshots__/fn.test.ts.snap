// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Builder SearchParams correctly converts URLSearchParams to object 1`] = `
Object {
  "__builder_editing__": "true",
  "abc": "klfdjklgfds",
  "builder.cachebust": "true",
  "builder.noCache": "true",
  "builder.overrides.037948e52eaf4743afed464f02c70da4": "037948e52eaf4743afed464f02c70da4",
  "builder.overrides.page": "037948e52eaf4743afed464f02c70da4",
  "builder.overrides.page:/": "037948e52eaf4743afed464f02c70da4",
  "builder.preview": "page",
  "preview_theme_id": "128854393017",
  "someotherValue": "jklsjfdal",
}
`;

exports[`Get Builder SearchParams correctly extracts all builder params from a query object 1`] = `
Object {
  "cachebust": "true",
  "noCache": "true",
  "overrides.037948e52eaf4743afed464f02c70da4": "037948e52eaf4743afed464f02c70da4",
  "overrides.page": "037948e52eaf4743afed464f02c70da4",
  "overrides.page:/": "037948e52eaf4743afed464f02c70da4",
  "preview": "page",
}
`;
